packages/SwingSet/src/kernel/kernel.js

```diff
@@ -158,6 +159,8 @@ export default function buildKernel(kernelEndowments) {
     log: [],
   };
 
+  const pendingMessageResults = new Map(); // kpid -> messageResult
```

Comment:
maybe this could only hold the `noteResolution` function, and not the entire `messageResult` object

Reply:
Done, though I kept it as a single method object rather than a free standing function in case we want to add other functionality later.